framework/src/play/mvc/ActionInvoker.java

```diff
@@ -484,8 +484,13 @@ public static Object[] getActionMethod(String fullAction) {
 
   public static Object[] getActionMethodArgs(Http.Request request, Session session, Method method) {
     String[] paramsNames = Java.parameterNames(method);
-    if (paramsNames == null && method.getParameterTypes().length > 0) {
-      throw new UnexpectedException("Parameter names not found for method " + method);
+
```

Comment:
since `paramsNames` is NEVER null this piece of code was never run. so I rewrote it so it now does what i guess it was intended to do.

when `-parameters` is not provided it will name arguments `arg0`, `arg1`, etc. i check for that here...




this is the biggest change i actually worked on in this PR, the rest were mainly IntelliJ's suggestions.

Reply:
we can also remove it btw... just lemme know.